### PR TITLE
wallet onboarding update

### PR DESCRIPTION
**Pull request by Jibz1**
https://user-images.githubusercontent.com/8676844/213801312-f38f90d3-ccb9-400b-af8d-1999ea0dcd4f.mov

### CODE REVIEW

**Comment by mystie711** [comment]
Looking fantastic overall @Jibz1! 

A couple small deltas i notice from Figma,
- in the onboarding launch screen update "...SUI network." to "...Sui network."
- please double check that the disabled "Continue ->" button color is correct one
- space between "I saved my..." and the CTA below
- space between "RECOVERY PHRASE" and text below it

**Comment by mystie711** [comment]
One more request, could we remove the pesky eye icon for wallet password field now that we have implemented the show/hide icon in the input field?
https://www.stefanjudis.com/snippets/how-to-hide-microsoft-edges-password-reveal-button/

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/initialize/backup/index.tsx:124]: I am going to remove the css module for this component. I did not want to break any existing design.

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/initialize/backup/index.tsx:107]: got it

File: apps/wallet/tailwind.config.js
Comment (Jibz1) [reply apps/wallet/tailwind.config.js:20]: Google it.  😄

File: apps/wallet/tailwind.config.js
Comment (Jibz1) [reply apps/wallet/tailwind.config.js]: done.

**Comment by mystie711** [comment]
@Jibz1 please double check the card shadow. The launch screen appears to have darker shadow than the rest. 
Would be a good idea to check the rest of the screens in the flow as well.

**Comment by Jibz1** [comment]
> @Jibz1 please double check the card shadow. The launch screen appears to have darker shadow than the rest.
> Would be a good idea to check the rest of the screens in the flow as well.

done.

File: apps/core/tailwind.config.js
Comment (Jordan-Mysten) [inline apps/core/tailwind.config.js:95]: I don't see this in the design system under typography?

File: apps/wallet/tailwind.config.js
Comment (Jordan-Mysten) [inline apps/wallet/tailwind.config.js:20]: If this is a color we're using can the figma be updated to include it? @mystie711

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/initialize/backup/index.tsx:74]: Branch all of the copy, not just the copy subset.

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/initialize/backup/index.tsx:57]: This set of classes seems like it's used a few times, could it be extracted into a component?

File: apps/wallet/src/ui/app/pages/welcome/index.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/welcome/index.tsx:32]: `11/12` seems odd, why that?

File: apps/wallet/src/ui/app/pages/initialize/select/index.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/initialize/select/index.tsx:40]: If you're doing a partial tailwind migration why not do it all?

File: apps/core/tailwind.config.js
Comment (Jibz1) [reply apps/core/tailwind.config.js:95]: I am removing this. It was a temp for the fonts icon. Once the icon PR lands, I will remove all fontsIcon

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/initialize/backup/index.tsx:57]: I plan to update and combine with the existing `CardLayout` component once I update that component to account for different styles

File: apps/wallet/src/ui/app/pages/welcome/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/welcome/index.tsx:32]: removing this

File: apps/wallet/src/ui/app/pages/initialize/select/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/initialize/select/index.tsx]: done

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/initialize/backup/index.tsx]: done

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/pages/initialize/backup/index.tsx:57]: I think we can just adapt the Cardlayout for the UI changes/requirements, (probably bg colors and text styles?) And make sure that views can adapt using the available space to avoid issues like the following

https://user-images.githubusercontent.com/10210143/211605870-d6db77db-2150-4839-9f51-8cd05c3279e4.mov

**Review by pchrysochoidis — CHANGES_REQUESTED** [review]

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/initialize/backup/index.tsx:132]: If we could make a checkbox component would be ideal! Probably it's easy just to extract it to it's one file.

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/shared/card-layout/index.tsx:15]: Can we remove the `w-popup-width` `h-popup-height` so we don't force card to have a specific size? I think page layout already does that

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/shared/card-layout/index.tsx:50]: why do we need this extra div?

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/shared/card-layout/index.tsx:63]: `text-2xl` should be fine

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/shared/card-layout/index.tsx:72]: `text-4xl` here as well

File: apps/wallet/src/ui/app/shared/field-label/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/shared/field-label/index.tsx:17]: Converting this to tailwind will allow to remove `FieldLabel.module.scss` as well and not have there unused classes

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/shared/card-layout/index.tsx:63]: To match the design as close as possible. text-2xl is about 2px off

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/shared/card-layout/index.tsx:72]: Same here

File: apps/wallet/src/ui/app/wallet/locked-page/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/wallet/locked-page/index.tsx:81]: just update the scss module instead of changing it to tailwind and just add the top margin there if necessary.

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/shared/card-layout/index.tsx:72]: Seems like we should figure out why we have inconsistent header sizes in the design, ideally we only use sizes that are reflected in the typography overview.

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/shared/card-layout/index.tsx:15]: This was mostly converted from the SCSS `max-width: sizing.$popup-width;
        max-height: sizing.$popup-height;`

Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/shared/card-layout/index.tsx:15]: I am not sure that the new design requires changes to the existing styling of the CardLayout. (maybe some additions) So why change the existing styling? The existing style tries to stretch/shrink depending the available space IIRC. Why converting it to tailwind has to change it?

Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/shared/card-layout/index.tsx:15]: >This was mostly converted from the SCSS max-width: sizing.$popup-width; max-height: sizing.$popup-height;

That is for when mode is box, and is max-width/height and this should move to the mode variant for box right?

File: apps/wallet/src/ui/app/pages/layout/Layout.module.scss
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/layout/Layout.module.scss:10]: Should we include the background conditionally? Currently it's always there and because the view is limited in popup size the content above covers the background.

**Comment by mystie711** [comment]
@Jibz1 just noticing a copy gaffe in Figma. Apologies.
(Just updated Figma with this detail.)

Would you please update the copy for both the paragraph and the checkbox line?
image

File: apps/wallet/src/ui/app/pages/layout/Layout.module.scss
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/layout/Layout.module.scss]: applying this to just `dynamic-size`

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/initialize/backup/index.tsx]: Agreed,  I will create a separate task for this.

File: apps/wallet/src/ui/app/pages/layout/Layout.module.scss
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/layout/Layout.module.scss]: done

File: apps/wallet/src/ui/app/wallet/locked-page/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/wallet/locked-page/index.tsx]: done

File: apps/wallet/src/ui/app/shared/field-label/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/shared/field-label/index.tsx]: done

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/shared/card-layout/index.tsx]: done

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/shared/card-layout/index.tsx]: done

File: apps/wallet/src/ui/app/shared/card-layout/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/shared/card-layout/index.tsx]: done

File: apps/wallet/src/ui/app/wallet/locked-page/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/wallet/locked-page/index.tsx]: > @Jibz1 just noticing a copy gaffe in Figma. Apologies. (Just updated Figma with this detail.)
> 
> Would you please update the copy for both the paragraph and the checkbox line? image

done

**Comment by Jibz1** [comment]
> @Jibz1 just noticing a copy gaffe in Figma. Apologies. (Just updated Figma with this detail.)
> 
> Would you please update the copy for both the paragraph and the checkbox line? image

@mystie711 done

**Review by pchrysochoidis — COMMENTED** [review]
Looks good in general, just a few comments.

There are a few places we need to add some margin: 

Screenshot 2023-01-17 at 15 44 18
Screenshot 2023-01-17 at 15 44 32

Screenshot 2023-01-17 at 15 40 37
Screenshot 2023-01-17 at 15 41 58
Screenshot 2023-01-17 at 15 42 59

There are a couple of places that custom arbitrary tailwind values are being used, I think the direction is to avoid them and if we need them add them to the theme.

Also the checkboxes should be converted to a component. (I am ok to do it in another PR, do we have a ticket for this?)

File: apps/wallet/src/ui/app/pages/initialize/create/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/initialize/create/index.tsx:23]: > Looks good in general, just a few comments.
> 
> There are a few places we need to add some margin:
> 
> Screenshot 2023-01-17 at 15 44 18
> 
> Screenshot 2023-01-17 at 15 44 32
> 
> Screenshot 2023-01-17 at 15 40 37
> 
> Screenshot 2023-01-17 at 15 41 58
> 
> Screenshot 2023-01-17 at 15 42 59
> 
> There are a couple of places that custom arbitrary tailwind values are being used, I think the direction is to avoid them and if we need them add them to the theme.
> 
> Also the checkboxes should be converted to a component. (I am ok to do it in another PR, do we have a ticket for this?)

Yeah, One of the reasons I kept some of the old SCSS. Let me address it

File: apps/wallet/src/ui/app/shared/field-label/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/shared/field-label/index.tsx:15]:
```suggestion
       <label className="flex flex-col flex-nowrap gap-2.5 mt-7.5 first:mt-0">
```

File: apps/wallet/src/ui/app/shared/field-label/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/shared/field-label/index.tsx:16]:
```suggestion
            <div className="ml-2">
```

File: apps/wallet/src/ui/app/shared/field-label/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/shared/field-label/index.tsx:21]:
```suggestion
```

File: apps/wallet/src/ui/app/pages/initialize/import/steps/StepOne.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/initialize/import/steps/StepOne.tsx:67]:
```suggestion
                            <Alert>{errors?.mnemonic}</Alert>
```

File: apps/wallet/src/ui/app/pages/initialize/shared/password-fields/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/initialize/shared/password-fields/index.tsx:22]:
```suggestion
                    <Alert>{errors.password}</Alert>
```

File: apps/wallet/src/ui/app/pages/initialize/shared/password-fields/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/initialize/shared/password-fields/index.tsx:28]:
```suggestion
                    <Alert>{errors.confirmPassword}</Alert>
```

File: apps/wallet/src/ui/app/pages/initialize/import/steps/StepOne.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/initialize/import/steps/StepOne.tsx:61]: Looks like `rounded-15` is already defined
Also checking the figma design `#EBECED` is gray-45. Maybe it changed or our tailwind config is wrong. Can we change here to border-gray-45 and fix tailwind config as well? 
Screenshot 2023-01-18 at 13 12 19

File: apps/wallet/src/ui/app/pages/initialize/backup/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/initialize/backup/index.tsx:88]: Same here for rounded and border color

**Review by pchrysochoidis — COMMENTED** [review]
In locked view the error message still need some margin. I added some comments about this and some small changes for arbitrary values usage. Should be good to merge after these changes.

**Review by pchrysochoidis — APPROVED** [review]
Before merging don't forget to change the error message for the password confirmation

File: apps/wallet/src/ui/app/pages/initialize/shared/password-fields/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/initialize/shared/password-fields/index.tsx:28]: got it

**Pull request merged**
